src/coreclr/jit/assertionprop.cpp

```diff
@@ -3177,33 +3177,78 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
 
     if (conValTree != nullptr)
     {
-        if (tree->OperIs(GT_LCL_VAR))
+        if (tree->OperIs(GT_LCL_VAR) && conValTree->OperIs(GT_CNS_VEC, GT_CNS_DBL, GT_CNS_INT))
```

Comment:
What about `GT_CNS_LNG`?

Reply:
Removed integers for now, they require more smart tunning apparently